Fabric.Identity.API/Program.cs

```diff
@@ -1,25 +1,70 @@
+using Fabric.Platform.Shared.Configuration.Docker;
+using Microsoft.AspNetCore;
 using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Serilog;
 
 namespace Fabric.Identity.API
 {
     public static class Program
     {
         public static void Main(string[] args)
         {
-            var appConfig = new Configuration.IdentityConfigurationProvider().GetAppConfiguration(Directory.GetCurrentDirectory());
-
-            var host = new WebHostBuilder()
-                .UseKestrel()
-                .UseContentRoot(Directory.GetCurrentDirectory())
-                .UseStartup<Startup>()
-                .UseApplicationInsights()
-                .UseIisIntegrationIfConfigured(appConfig)
-                .UseUrls("http://*:5001")
+            var loggerConfiguration = new LoggerConfiguration();
+
+            //Create a builder just so that we can read the app insights instrumentation key from the config
+            var configuration = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddJsonFile("appsettings.json", optional: false, reloadOnChange: true)
+                .AddJsonFile($"appsettings.{Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") ?? "Production"}.json", optional: true)
                 .Build();
 
-            host.Run();
+            var certificateService = IdentityConfigurationProvider.MakeCertificateService();
+            var decryptionService = new DecryptionService(certificateService);
+            var appConfig = new IdentityConfigurationProvider(configuration).GetAppConfiguration(decryptionService);
+
+            LogFactory.ConfigureTraceLogger(loggerConfiguration, appConfig.ApplicationInsights);
+
+            Log.Logger = loggerConfiguration.CreateLogger();
+
+            try
+            {
+                Log.Information("Starting web host");
+                CreateWebHostBuilder(args).Build().Run();
+
+                Environment.Exit(0);
+            }
+            catch (Exception ex)
+            {
+                Log.Fatal(ex, "Host terminated unexpectedly");
+
+                Environment.Exit(1);
+            }
+            finally
+            {
+                Log.CloseAndFlush();
+            }
         }
+
+        public static IWebHostBuilder CreateWebHostBuilder(string[] args) =>
+            WebHost.CreateDefaultBuilder(args)
+                .ConfigureAppConfiguration((hostContext, config) =>
+                {
+                    config.AddDockerSecrets(typeof(IAppConfiguration));
+                    config.SetBasePath(hostContext.HostingEnvironment.ContentRootPath);
+                })
+                .UseUrls("http://*:5001")
+                .ConfigureKestrel((context, options) =>
+                {
+                    // Set properties and call methods on options
+                })
+                .UseSerilog()
+                .UseStartup<Startup>();
```

Comment:
Is all of this just auto added after the .NET Core upgrade? I'm not sure if all of it needed/ok to have. Maybe it's all fine.

Reply:
.Net Core 2.2 does things a little differently, and the upgrade guides say to do this. That being said, most of what is in the Main method is due to the Serilog upgrades.